### DISCUSSION TITLE
Migrate from miniforge3 (x86_64) to miniforge3 (arm) on mac M1

### DISCUSSION THREAD

**Discussion by shahpnmlab** [conda-forge/miniforge#442] [Q&A]
I have being using miniforge x86_64 under rosetta on my m1 mac for the last few years and only recently installed the arm compatible version of miniforge. is there a non-destructive way of migrating my environments from the x86_64 version to the arm64 version?

**Comment by jaimergp** [comment]
You can try using the `subdir` key in your `.condarc` in some smart way:

- Annotate all your envs with a `.condarc` and `subdir: osx-64` so they stay how they are.
- Then switch to `subdir: osx-arm64` in your _user_ `.condarc` (`~/.condarc`).

That is, assuming you want to preserve your environments. If you change the default `subdir` only in your user configuration, then you might end up with a successful transition, but I can see how some envs cannot be solved or involve surprising regressions.
